consensus/src/state_computer.rs

```diff
@@ -193,35 +191,37 @@ impl StateComputer for ExecutionProxy {
             .await
             .expect("Failed to send async state sync notification");
 
-        // If there are no blocks, epoch and round will be invalid.
-        // TODO: is this ever the case? why?
-        if skip_clean {
-            return Ok(());
-        }
-        payload_manager
-            .notify_commit(LogicalTime::new(latest_epoch, latest_round), payloads)
-            .await;
+        *latest_logical_time = logical_time;
+        payload_manager.notify_commit(logical_time, payloads).await;
         Ok(())
     }
 
     /// Synchronize to a commit that not present locally.
     async fn sync_to(&self, target: LedgerInfoWithSignatures) -> Result<(), StateSyncError> {
-        let _guard = self.write_mutex.lock().await;
+        let mut latest_logical_time = self.write_mutex.lock().await;
+        let logical_time =
+            LogicalTime::new(target.ledger_info().epoch(), target.ledger_info().round());
 
         // Before the state synchronization, we have to call finish() to free the in-memory SMT
         // held by BlockExecutor to prevent memory leak.
         self.executor.finish();
 
+        // The pipeline phase already committed beyond the target synced round, just return.
+        if *latest_logical_time >= logical_time {
+            warn!(
```

Comment:
This is a bit of a nit. I thought we established that `==` is possible even without this race. It makes sense to return right away in this case (nothing more to do), but maybe only warn in the `>` case?

Reply:
it's similarly unexpected, just state sync would not return error previously. it's expected we only sync to future round